src/Servers/Kestrel/Core/src/Internal/HttpConnection.cs

```diff
@@ -58,22 +60,29 @@ public HttpConnection(HttpConnectionContext context)
                 {
                     case HttpProtocols.Http1:
                         // _http1Connection must be initialized before adding the connection to the connection manager
-                        requestProcessor = _http1Connection = new Http1Connection<TContext>(_context);
+                        requestProcessor = _http1Connection = new Http1Connection<TContext>((HttpConnectionContext)_context);
                         _protocolSelectionState = ProtocolSelectionState.Selected;
                         break;
                     case HttpProtocols.Http2:
                         // _http2Connection must be initialized before yielding control to the transport thread,
                         // to prevent a race condition where _http2Connection.Abort() is called just as
                         // _http2Connection is about to be initialized.
-                        requestProcessor = new Http2Connection(_context);
+                        requestProcessor = new Http2Connection((HttpConnectionContext)_context);
+                        _protocolSelectionState = ProtocolSelectionState.Selected;
+                        break;
+                    case HttpProtocols.Http3:
+                        // _http2Connection must be initialized before yielding control to the transport thread,
+                        // to prevent a race condition where _http2Connection.Abort() is called just as
+                        // _http2Connection is about to be initialized.
```

Comment:
I think the copied comments was already a bit out of date (I think it should have been referencing `_requestProcessor` instead of `_http2Connection`) and it makes even less sense here.